manylinux_2_31/riscv64/Dockerfile

```diff
@@ -0,0 +1,237 @@
+
+FROM ubuntu:22.04 AS toolchain
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y \
+    automake \
+    bison \
+    bzip2 \
+    ca-certificates \
+    cmake \
+    curl \
+    file \
+    flex \
+    g++ \
+    gawk \
+    gdb \
+    git \
+    gperf \
+    help2man \
+    libncurses-dev \
+    libssl-dev \
+    libtool-bin \
+    lzip \
+    make \
+    ninja-build \
+    patch \
+    pkg-config \
+    python3 \
+    rsync \
+    sudo \
+    texinfo \
+    unzip \
+    wget \
+    xz-utils \
+    libssl-dev \
+    libffi-dev
+
+# Install crosstool-ng
+# Backport https://github.com/crosstool-ng/crosstool-ng/pull/{2315,2316} to fix build
+RUN curl -Lf https://github.com/crosstool-ng/crosstool-ng/archive/crosstool-ng-1.27.0.tar.gz | tar xzf - && \
+    cd crosstool-ng-crosstool-ng-1.27.0 && \
+    curl -Lf https://github.com/crosstool-ng/crosstool-ng/pull/2315.patch | patch -Np1 && \
+    curl -Lf https://github.com/crosstool-ng/crosstool-ng/pull/2316.patch | patch -Np1 && \
+    ./bootstrap && \
+    ./configure --prefix=/usr/local && \
+    make -j4 && \
+    make install && \
+    cd .. && rm -rf crosstool-ng-*
+
+COPY .config /tmp/toolchain.config
+
+# Build cross compiler
+RUN mkdir build && \
+    cd build && \
+    cp /tmp/toolchain.config .config && \
+    export CT_ALLOW_BUILD_AS_ROOT_SURE=1 && \
+    ct-ng build.2 || { tail -n 500 build.log; exit $ERRCODE; } && \
+    cd .. && \
+    rm -rf build
+
+FROM ubuntu:22.04
+
+# Copy cross toolchain
+COPY --from=toolchain /usr/riscv64-unknown-linux-gnu /usr/riscv64-unknown-linux-gnu
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV PATH=$PATH:/usr/riscv64-unknown-linux-gnu/bin
+
+ENV CC_riscv64_unknown_linux_gnu=riscv64-unknown-linux-gnu-gcc \
+    AR_riscv64_unknown_linux_gnu=riscv64-unknown-linux-gnu-ar \
+    CXX_riscv64_unknown_linux_gnu=riscv64-unknown-linux-gnu-g++
+
+ENV TARGET_CC=riscv64-unknown-linux-gnu-gcc \
+    TARGET_AR=riscv64-unknown-linux-gnu-ar \
+    TARGET_RANLIB=riscv64-unknown-linux-gnu-ranlib \
+    TARGET_CXX=riscv64-unknown-linux-gnu-g++ \
+    TARGET_READELF=riscv64-unknown-linux-gnu-readelf \
+    TARGET_SYSROOT=/usr/riscv64-unknown-linux-gnu/riscv64-unknown-linux-gnu/sysroot/ \
+    TARGET_C_INCLUDE_PATH=/usr/riscv64-unknown-linux-gnu/riscv64-unknown-linux-gnu/sysroot/usr/include/
+
+ENV CARGO_BUILD_TARGET=riscv64-unknown-linux-gnu
```

Comment:
just noticed, the rust target should be [riscv64gc-unknown-linux-gnu](https://doc.rust-lang.org/rustc/platform-support/riscv64gc-unknown-linux-gnu.html#riscv64gc-unknown-linux-gnu)?

Reply:
Fixed in 474db27e0a710ffd8b4a6006cd7914db538bd196

Reply:
> just noticed, the rust target should be [riscv64gc-unknown-linux-gnu](https://doc.rust-lang.org/rustc/platform-support/riscv64gc-unknown-linux-gnu.html#riscv64gc-unknown-linux-gnu)?

Oops. Nice catch!